src/encode.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+  Copyright (c) 2014-2016 DataStax
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#include "encode.hpp"
+#include "utils.hpp"
+
+namespace cass {
+
+static Buffer encode_internal(CassDuration value, bool with_length) {
+  // Each duration attribute needs to be converted to zigzag form. Use an array to make it
+  // easy to do the same encoding ops on all three values.
+  uint64_t zigzag_values[3];
+
+  // We need varint sizes for each attribute.
+  size_t varint_sizes[3];
```

Comment:
What do you think about changing the name of these to  "vint" instead of "varint"? "varint" is already a Cassandra type.